### ISSUE TITLE
`bun test` with globs not detecting all tests

### ISSUE THREAD

**Issue by webpro** [oven-sh/bun#10353]
### What version of Bun is running?

1.1.3+2615dc742

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

Given files like so:

```
│   └── test
│       ├── module.test.ts
│       ├── script.test.ts
│       ├── dir
│       │   ├── file.test.ts
│       └── config
│           └── config.test.ts
│
```

Then this command:

```
bun test test/**/*.test.ts
```

Does not run tests in `test/module.test.ts` and `test/script.test.ts`. It does run the other deeper tests.

### What is the expected behavior?

It should also match `test/*.test.ts`

### What do you see instead?

It matches only files in sub directories.

This is a local workaround:

```
bun test test/*.test.ts test/**/*.test.ts
```

(edited just to match extensions)

**Comment by zackradisic** [comment]
This looks like the pattern is being expanded by your shell (and not bun) and your shell is not configured to handle the `globstar` option

**Comment by webpro** [comment]
No, my shell is configured properly (e.g. with `ls` it expands fine too).

But now I gather that the issue only arises when running through `package.json#scripts`, not when running from the command-line directly. Sorry for not mentioning that right away.

**Comment by i-void** [comment]
I'm also affecting from this. +1
